util/src/main/java/io/kubernetes/client/util/credentials/TokenFileAuthentication.java

```diff
@@ -0,0 +1,59 @@
+package io.kubernetes.client.util.credentials;
+
+import io.kubernetes.client.openapi.ApiClient;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Instant;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class TokenFileAuthentication implements Authentication, Interceptor {
```

Comment:
i presume using `Interceptor` to hijack the header is a bit overkilling.. the previous approach by `AccessTokenAuthentication` is setting tokens by `((ApiKeyAuth) auth).setApiKey(apiKey);` in the `provide` call, wdyt if we do the token-reloading similarly?

Reply:
the current openapi doesn't have reloading logic in `ApiKeyAuth` which will only be set once in this java client.

we need something similar to this change https://github.com/OpenAPITools/openapi-generator/pull/6036 (it might take much longer time) to implement reloading.

Reply:
> the current openapi doesn't have reloading logic in ApiKeyAuth which will only be set once in the initialization of in-cluster client.

convinced, can you leave a TODO comment to reference OpenAPITools/openapi-generator#6036 so that we can converge in the future?

Reply:
yes, done